CDTDatastore/HTTP/CDTRequestLimitInterceptor.m

```diff
@@ -0,0 +1,46 @@
+//
+//  CDTRequestLimitInterceptor.m
+//  CDTDatastore
+//
+//  Created by tomblench on 23/06/2016.
+//  Copyright © 2016 IBM. All rights reserved.
+//
+
+#import "CDTRequestLimitInterceptor.h"
+
+@interface CDTRequestLimitInterceptor ()
+
+@property NSTimeInterval sleep;
+
+
+@end
+
+@implementation CDTRequestLimitInterceptor
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        self.sleep = 0.25; // 250ms
+    }
+    return self;
+}
+
+/**
+ * Interceptor to retry after an exponential backoff if we receive a 429 error
+ */
+- (CDTHTTPInterceptorContext *)interceptResponseInContext:(CDTHTTPInterceptorContext *)context
+{
+    if (context.response.statusCode == 429) {
+        
+        CDTLogInfo(CDTTD_REMOTE_REQUEST_CONTEXT, @"429 error code received. Will retry in %f seconds.", self.sleep);
```

Comment:
I think this log message will be misleading on the last attempt since I think it will get logged anyway declaring that there will be a retry in x seconds, but the outer retry control at https://github.com/cloudant/CDTDatastore/blob/master/CDTDatastore/HTTP/CDTURLSessionTask.m#L201 will prevent another request being made. I guess that was one advantage of having the number of retries available in the Java interceptor that I had forgotten about when we discussed it.

Further to that I wonder if it would be helpful to include the "too many requests" to save people needing to look it up if they are unfamiliar with the status code.